tests/components/climate/test_generic_thermostat.py

```diff
@@ -52,25 +59,41 @@ def tearDown(self):  # pylint: disable=invalid-name
     def test_setup_missing_conf(self):
         """Test set up heat_control with missing config values."""
         config = {
-            'name': 'test',
-            'target_sensor': ENT_SENSOR
+            'name': 'test'
         }
         with assert_setup_component(0):
             setup_component(self.hass, 'climate', {
                 'climate': config})
 
-    def test_valid_conf(self):
+    def test_setup_valid_conf(self):
         """Test set up generic_thermostat with valid config values."""
         self.assertTrue(
             setup_component(self.hass, 'climate',
                             {'climate': {
                                 'platform': 'generic_thermostat',
                                 'name': 'test',
-                                'heater': ENT_SWITCH,
+                                'heater_control': ENT_SWITCH_HEAT,
                                 'target_sensor': ENT_SENSOR
                                 }})
         )
 
+    @mock.patch('logging.Logger.error')
```

Comment:
Instead of this, use the fixture `caplog`. To use fixtures you'll have to rewrite your test to be a standalone function.

```python
async def test_setup_valid_conf_with_error(hass, caplog):
    await async_setup_component(…)
    assert 'Your error text' in caplog.text
```

Reply:
Usually the fact that we log an error is not so interesting. It's more interesting to see if the entity was still setup despite an error being logged.

Reply:
I will look into rewriting the test.